src/seedu/addressbook/storage/AddressBookEncoder.java

```diff
@@ -0,0 +1,67 @@
+package seedu.addressbook.storage;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.tag.Tag;
+
+/**
+ * Encodes the AddressBook object into a data file for storage.
+ */
+public class AddressBookEncoder {
+
+    /**
+     * Encodes all the Persons in the AddressBook to the data file for storage.
+     *
+     * @param toSave AddressBook to encode.
+     * @return encoded strings.
+     */
+    public static List<String> encodeAddressBook(AddressBook toSave) {
+        return encodePersonsToStrings(toSave.getAllPersons());
+    }
+
+    /**
+     * Encodes persons into list of decodable and readable string representations.
+     *
+     * @param persons to be encoded.
+     * @return encoded strings.
+     */
+    private static List<String> encodePersonsToStrings(UniquePersonList persons) {
+        final List<String> encodedPersons = new ArrayList<>();
+        for (Person person : persons) {
+            encodedPersons.add(encodePersonToString(person));
+        }
```

Comment:
Loop can be simplified:
`persons.forEach(person -> encodedPersons.add(encodePersonToString(person)));`

Reply:
Likewise, streams is only being used in AB4. 
Is there a reason for that, or the change was just not brought down to the lower levels?

Reply:
Changes are not brought down to the lower levels because it would be 2x more work for contributors and reviewers. (It's 2x more work because AB-2, AB-3 and AB-4 all differ in subtle ways, e.g. some comments missing or added in some places or something implemented in a slightly different manner)

The current plan is to fork off a new version of AB-3 (and then AB-2) from the AB-4 code base. That way, they will share a common ancestor and we can then use merges to propagate changes from one code base to another.